tests/search/search_test.php

```diff
@@ -0,0 +1,234 @@
+        $search = \testable_core_search::instance();
+
+        $cmstype = new cms_types();
+        $cmstype->set('name', 'Overview')
+            ->set('idnumber', 'overview')
+            ->set('title_mustache', 'Overview');
+        $cmstype->save();
+        $fieldcategory = self::getDataGenerator()->create_custom_field_category([
+            'name' => 'Other fields',
+            'component' => 'mod_cms',
+            'area' => 'cmsfield',
+            'itemid' => $cmstype->get('id'),
+        ]);
+        $field = self::getDataGenerator()->create_custom_field([
+            'name' => 'Overview',
+            'shortname' => 'overview',
+            'type' => 'text',
+            'categoryid' => $fieldcategory->get('id'),
+        ]);
+        $this->cmstype = $cmstype;
+        $this->fieldcategory = $fieldcategory;
+        $this->field = $field;
+    }
+
+    /**
+     * Test search enabled.
+     *
+     * @return void
+     */
+    public function test_search_enabled(): void {
+        $searcharea = \core_search\manager::get_search_area($this->cmsareaid);
+        list($componentname, $varname) = $searcharea->get_config_var_name();
+
+        // Enabled by default once global search is enabled.
+        $this->assertTrue($searcharea->is_enabled());
+
+        set_config($varname . '_enabled', 0, $componentname);
+        $this->assertFalse($searcharea->is_enabled());
+
+        set_config($varname . '_enabled', 1, $componentname);
+        $this->assertTrue($searcharea->is_enabled());
+    }
+
+    /**
+     * Indexing mod cms contents.
+     *
+     * @return void
+     */
+    public function test_get_document_recordset(): void {
+        global $DB;
+
+        // Returns the instance as long as the area is supported.
+        $searcharea = \core_search\manager::get_search_area($this->cmsareaid);
+        $this->assertInstanceOf('\mod_cms\search\cmsfield', $searcharea);
+
+        $course = self::getDataGenerator()->create_course();
+
+        // Name for cms is coming from "title_mustache" in cms_type.
+        $generator = self::getDataGenerator()->get_plugin_generator('mod_cms');
+        $record = new \stdClass();
+        $record->course = $course->id;
+        $record->customfield_overview = 'Test overview text 1';
+        $record->typeid = $this->cmstype->get('id');
+        $generator->create_instance_with_data($record);
+
+        $record = new \stdClass();
+        $record->course = $course->id;
+        $record->customfield_overview = 'Test overview text 2';
+        $record->typeid = $this->cmstype->get('id');
+        $generator->create_instance_with_data($record);
+
+        // All records.
+        $recordset = $searcharea->get_document_recordset();
+        $this->assertTrue($recordset->valid());
+        $this->assertEquals(2, iterator_count($recordset));
+        $recordset->close();
+
+        // The +2 is to prevent race conditions.
```

Comment:
Can you please explain a bit more about this comment? What race conditions we are trying to avoid?

Reply:
It's just copied from other plugin actually.
I updated the comment.
It's just change the time to future to make sure it's empty as there is no record after that time.